SynthDet/Assets/Scripts/Randomizers/ForegroundObjectMetricReporter.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using SynthDet.RandomizerTags;
+using Unity.Mathematics;
+using UnityEngine;
+using UnityEngine.Perception.GroundTruth;
+using UnityEngine.Perception.Randomization.Randomizers;
+using Object = UnityEngine.Object;
+
+namespace SynthDet.Randomizers
+{
+    /// <summary>
+    /// Reports metrics for rotation, world position, scale, and label id of objects that are tagged with <see cref="ForegroundObjectMetricReporterTag"/>.
+    /// </summary>
+    [Serializable]
+    [AddRandomizerMenu("Perception/My Foreground Object Metric Reporter")]
+    public class ForegroundObjectMetricReporter : Randomizer
```

Comment:
Couple questions on this:
* Why not use 3d bounding boxes to capture this information?
* Since this is specifically for foreground objects, why not have this code in the ForegroundObjectPlacementRandomizer?

Reply:
<resolved offline>
Summary: The Randomizer workflow provides a powerful way of coordinating timing of code execution within the Scenario, and thus makes it easy to make sure the reporting is done after all other Randomizers are done. Thus, for the time being, we will keep this pattern for metric reporters. This may be iterated on in future versions.